index.js

```diff
@@ -0,0 +1,152 @@
+$(document).ready(() => {
+  listTrips();
+  loadTripDetails();
+});
+
+const getURL = "https://trektravel.herokuapp.com/trips";
+
+const reportStatus = message => {
+  $("#status-message").html(message);
+};
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p><ul>`;
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      content += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  content += "</ul>";
+  reportStatus(content);
+};
+
+// This loads the trip
+const getTrips = () => {
+  reportStatus("Loading trips...");
+
+  const tripList = $("#trip-list");
+  tripList.empty();
+
+  axios
+    .get(getURL)
+    .then(response => {
+      reportStatus(`Successfully loaded ${response.data.length} trips`);
+      response.data.forEach(trip => {
+        tripList.append(`<li class='${trip.id}'>${trip.name}</li>`);
+      });
+    })
+    .catch(error => {
+      reportStatus(
+        `Encountered an error while loading trips: ${error.message}`
+      );
+      console.log(error);
+    });
+};
+
+// This gets the trip details given the id as a parameter
+const detailsTrips = id => {
+  reportStatus("Loading details...");
+
+  const tripDetailsList = $("#trip-details");
+  tripDetailsList.empty();
+
+  axios
+    .get(getURL)
+    .then(response => {
+      reportStatus(`Successfully loaded ${response.data.length} trips`);
+      // callback function which takes the id passed to it and finds the trip associated with that id
+      const trip = response.data.find(item => {
+        // returns true when that id matches the id passed to this method
+        return item.id.toString() === id;
```

Comment:
This implementation works, but it's a little odd. It looks like your strategy is:

1. Make a call to the endpoint for the full list of trips
1. Search through the list for the trip that matches the ID for this trip

However, the API has an endpoint to get details for a single trip - since you've got the ID, it might make sense to call that instead and have less network traffic. That would also get you real data for the `about` section.